update/Pos/Update/Poll/PollState.hs

```diff
@@ -48,19 +53,37 @@ data PollState = PollState
       -- | All stakeholders who made proposals in the current epoch
     , _psEpochProposers     :: !(HashSet StakeholderId)
       -- | All applications in use and their latest (confirmed) versions
-    , _psConfirmedBVs       :: !(HashMap ApplicationName NumSoftwareVersion)
+    , _psConfirmedBVs       :: !(HM.HashMap ApplicationName NumSoftwareVersion)
       -- | All proposed software versions and their state
-    , _psConfirmedProposals :: !(HashMap SoftwareVersion ConfirmedProposalState)
+    , _psConfirmedProposals :: !(HM.HashMap SoftwareVersion ConfirmedProposalState)
       -- | All update proposals and their states
-    , _psActiveProposals    :: !(HashMap UpId ProposalState)
+    , _psActiveProposals    :: !(HM.HashMap UpId ProposalState)
       -- | Update proposals for each application
-    , _psDelActivePropsIdx  :: !(HashMap ApplicationName (HashSet UpId))
+    , _psDelActivePropsIdx  :: !(HM.HashMap ApplicationName (HashSet UpId))
       -- | Slotting data for this node
     , _psSlottingData       :: !SlottingData
       -- | Mapping between epochs and their richmen stake distribution
-    , _psFullRichmenData    :: !(HashMap EpochIndex FullRichmenData)
+    , _psFullRichmenData    :: !(HM.HashMap EpochIndex FullRichmenData)
       -- | Mapping between epochs and stake of each of the epoch's slot's block issuer
-    , _psIssuersStakes      :: !(HashMap EpochIndex IssuersStakes)
+    , _psIssuersStakes      :: !(HM.HashMap EpochIndex IssuersStakes)
     } deriving (Show, Eq)
 
 makeLenses ''PollState
+
+modifyPollState :: PollModifier -> PollState -> PollState
+modifyPollState PollModifier {..} PollState {..} =
+    let insAndDel pm hm =
+            foldl' (flip (uncurry HM.insert)) (foldl' (flip HM.delete) hm deletes) inserts
+          where inserts = insertions pm
+                deletes = deletions pm
+    in PollState (insAndDel pmBVs _psBlockVersions)
+                 _psAdoptedBV
+                 _psSoftwareVersion
+                 _psEpochProposers
```

Comment:
Suppose there is `a :: A` in `PollState` and `b :: Maybe A` in `PollModifier`. If `b` is `Just c`, result is `c`, otherwise result is `a`.

Semantics of `Maybe`s in `PollModifier` is that if it's `Nothing`, then it preserves old value, otherwise it sets new value.